### PR TITLE
Popcorn player

### PR DESCRIPTION
**Pull request by traceypooh**
New player / "embed.html" mode.  It allows quick, simply way to play/view a popcorn project via a JSON URL.
The editor / "editor.html" mode can now save a project to JSON in a new archive.org item (which can in turn be played by this "new" player).   

The prior way to embed popcorn was via hardcoded markup in html files
using JS eval() to setup the timeline, project, and tracks area.

Now we simply use a new "savedDataUrl" arg to load the popcorn project
and inject it into an initial Popcorn object.

### CODE REVIEW

**Comment by traceypooh** [comment]
oh, i should add, this can be seen in action (dev not full production) now at:
https://www-tracey.archive.org/details/Tracey_pooh-politicalAdscheckPlease
and you can click the "remix" button to head to the editor, etc.
(I used the editor [SAVE] button to make that item)

**Comment by Nolski** [comment]
This is AWESOME!!! I'm a bit swamped with a paper due tomorrow morning but I'll be sure to review and merge if it all looks good tomorrow.

**Comment by traceypooh** [comment]
woohoo!!  thanks so much and g'luck w/ the paper.
can't wait to hear your thoughtses...

**Comment by bvibber** [comment]
Nice! I'll take a peek later...

**Comment by jbraun27** [comment]
I've been using a modified a version of embed.js from the old popcorn-maker project for the past few years. Embed.js moved over from the popcorn.webmaker.org project, but the embed.html wasn't. It's here: [https://github.com/mozilla/popcorn.webmaker.org/blob/master/views/embed.html](url) 

My requirement was to add controls for users watching popcorn sequences, to tag up video that they're watching, and save those in-out points to an EDL format, and an SQL database. The great thing about embed.js/html is that it strips out a lot of overhead, and modules that aren't necessary just to play popcorn sequences. (See lines 19-24 of src/embed.js) 

This is great to include it into popcorn-editor project. Thanks @traceypooh I talked with Mike about this at least a year ago and he gave me some great advice! (Thanks @Nolski ) And thank you all for resurrecting this project!!! It needs to be kept alive, and yes @brion, an ES6 refactoring would be amazing if someone has the time to do so.

File: PopcornEditor/src/embed.js [hunk 1]
```diff
@@ -446,12 +448,33 @@ function init() {
...
+
+      var json = '';
+      if ( !config.savedDataUrl ) {
+        config.savedDataUrl = home + 'templates/basic/projects/archive.json';
+      }
+
+      // do a BLOCKING ajax load of the project JSON
+      config.debug  &&  console.log( 'savedDataUrl:', config.savedDataUrl );
+      jQuery.ajax({url: config.savedDataUrl,
+                   async: false,
```
Comment (Nolski) [inline PopcornEditor/src/embed.js]: Why does this have to be blocking? Can't we just initialize everything that needs the json from this call in the callback?

**Comment by Nolski** [comment]
All in all, this looks clean and good to merge. I'll allow @brion to take a look and if it all looks good to him as well, we can merge it in :shipit:

**Comment by traceypooh** [comment]
(quick aside -- for years i admired/loved the popcorn team ("take me with you!") and last night, while thinking about working with and learning from some of my favorite, nicest, smartest people on the planet i know (you three!) it made me realize "omg!  we _are_ that team" now 😂  )

thanks, @Nolski  (and TIA @brion) for looking.

yes, absolutely right on that -- prolly should find a better way than blocking ajax (and hook in some kind of callback and/or promise solution -- was a bit unsure about the best way to go about that).  we simply can't continue the JS execution to setup the player until we have that project JSON (make sense) so that's why i did that (prior webmaker code would have the JS "pre-injected" directly into the <head> of the  markup on a given (one per project) project htm page, etc.

btw, should have mentioned, anything marked "xxxp" is a quickly re-findable note (esp.) for me and a kind of "future promise" to make sure it's updated/fixed/resolved before this code goes into production (but allows us to continue fwd motion and idea sharing as we figure out where this codebase goes).  so i should really have put a "xxxp" on that blocking JSON call!

finally, you prolly noticed i (started?) using jQuery more formally in the JS instead of the more typical popcorn lib way.  how do we feel about that?  seem ok?  i'm much more fluent and comfortable w/ the jQuery framework/stack and (imho) find it more concise, industry std, grokable.  but i'm flexible on that (and all things).

really great working with y'all!
we had a huge excited buzz about this at our open friday lunch here as we now have a dev setup of a working popcorn player, editor and "save my project to new IA item" working end-to-end at a basic level.

i'm so thrilled esp. by the major legwork you did @Nolski -- it's great to feel tall standing on top of giants!   8-)  go team!

**Comment by traceypooh** [comment]
@jbraun27 thanks so much for the comments and enthusiasm.  i'm thrilled this may be a helpful update for you and others!

**Comment by traceypooh** [comment]
OK, minor updates to get _most_ features of _most_ prior projects working.  so projects with things like maps, popups, etc. are working or almost entirely working now.
eg:
http://www-tracey.archive.org/pop/embed.html?savedDataUrl=templates/basic/projects/stop-and-frisk.json

**Comment by traceypooh** [comment]
hey @brion any objections if i merge my own PR, _bad form_ et al?
only a couple minor fixmes fixed since may -- and it's sort of **<COUGH>** in production(-ish, quietly) now at https://archive.org/pop

I now have an upcoming (much smaller) feature PR for reading payloading hack of full project JSON into a JPEG/EXIF as a way to dodge full archive.org login requirements to "save" a project.   that is, we could provide a simple service on archive.org to take a project's JSON/EDL and inject it into a project's thumbnail so it is "self-referential" and trying to load a project/JSON from a .jpg file

eg:
https://archive.org/pop/embed.html?savedDataUrl=https://archive.org/~tracey/pope/injected.jpg
from/gist:
https://archive.org/~tracey/pope/

also brion -- LMK if we should add you to a new slack popcorn hacker channel that ben moskowitz setup!

**Pull request merged**